protocol-units/da/movement/celestia/light-node/src/v1/sequencer.rs

```diff
@@ -513,10 +513,11 @@ pub mod block {
 
 	impl Splitable for WrappedBlock {
 		fn split(self, factor: usize) -> Result<Vec<Self>, anyhow::Error> {
+			let namespace = self.blob.namespace;
 			let split_blocks = self.block.split(factor)?;
 			let mut wrapped_blocks = Vec::new();
 			for block in split_blocks {
-				let wrapped_block = WrappedBlock { block, blob: self.blob.clone() };
+				let wrapped_block = WrappedBlock::try_new(block, namespace)?;
```

Comment:
I wonder if anything in block creation is actually fallible.
The block to wrap is supposed to be correct by construction, and it would be unexpected for the BCS encoder to fail. Or are there size restrictions on encoded bytes, for example?

Reply:
I don't think `bcs::to_bytes` is actually fallible. Not sure about `zstd`.